docker-compose/docker-compose.yml

```diff
@@ -81,12 +82,15 @@ services:
       - ~/tmp/openwhisk/local.env     # generated during make setup
     environment:
       COMPONENT_NAME: invoker
-      SERVICE_NAME: invoker0
+      INVOKER_NAME: invoker0
```

Comment:
2 things: 
- it's fine to introduce redis usage from invoker if that is the goal, but it's not required if you still pass the instance id at the invoker; for compose env it may be simpler to NOT rely on redis from invoker so there is no risk of startup timing issues (if invoker starts first and redis is NOT listening yet)
- if you DO want to rely on redis from invoker in compose env, I would give the INVOKER_NAME something that is unlike the instance id naming approach, e.g. `INVOKER_NAME: compose-invoker-1`, or similar

Reply:
I think for compose we don't need to rely on Redis.

Reply:
Given that Redis is not a dependency anymore, do you still think we should give it a distinct name ?

Reply:
I think if INVOKER_NAME is specified, it will require redis? To NOT require redis, I think requires using the args on invoker to specify the instance id

Reply:
```[ERROR] [??] [Config] required property invoker.name still not set
[ERROR] [??] [Invoker] Bad configuration, cannot start.```

Reply:
Maybe this https://github.com/apache/incubator-openwhisk/blob/45e392e13fc7cd659e2a29b20ab070288412e200/core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala#L62 
should be changed to:
```Map(invokerName -> "")```
@rabbah @dgrove-oss I assume the expected behavior is that NEITHER redis host NOR invoker name should be required configs?

Reply:
I've been traveling so I am a bit behind on actually coding.  

When I get a couple of hours and work on https://github.com/apache/incubator-openwhisk/issues/2872, my intent is to make the invokerName envvar optional (since it could be provided on the command line, or not at all if an invoker id is provided either via the command line or the environment).

Reply:
> make the invokerName envvar optional

see https://github.com/apache/incubator-openwhisk/pull/2902 as a first step